stingray/lightcurve.py

```diff
@@ -167,8 +167,19 @@ def __init__(self, time, counts, err=None, input_counts=True,
 
         self.mjdref = mjdref
         self.time = np.asarray(time)
+        dt_array = np.diff(np.sort(self.time))
+        dt_array_unsorted = np.diff(self.time)
+        unsorted = np.any(dt_array_unsorted < 0)
+
         if dt is None:
-            self.dt = np.median(self.time[1:] - self.time[:-1])
+            if unsorted:
+                logging.warning("The light curve is unordered! This may cause "
+                                "unexpected behaviour in some methods! Use "
+                                "sort() to order the light curve in time and "
+                                "check that the time resolution `dt` is "
+                                "calculated correctly!")
+
+            self.dt = np.median(dt_array)
```

Comment:
There are numerical errors arising when we do that. We had to set `dt` to some average along the array, because the propagated error of 0.9999999 instead of 1 over a million elements is not negligible.